modules/core-tests/src/test/scala/trace4cats/QueuedSpanCompleterSpec.scala

```diff
@@ -0,0 +1,74 @@
+package trace4cats
+
+import cats.effect.IO
+import cats.effect.kernel.Ref
+import cats.effect.std.Random
+import cats.effect.testkit.TestInstances
+import cats.effect.unsafe.implicits.global
+import cats.syntax.all._
+import fs2.Chunk
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+import trace4cats.test.ArbitraryInstances._
+
+import scala.concurrent.duration._
+
+class QueuedSpanCompleterSpec extends AnyFlatSpec with Matchers with TestInstances with ScalaCheckDrivenPropertyChecks {
+  behavior.of("QueuedSpanCompleter")
+
+  def delayedExporter(ref: Ref[IO, Int]): SpanExporter[IO, Chunk] = new SpanExporter[IO, Chunk] {
+    def exportBatch(batch: Batch[Chunk]): IO[Unit] = IO.sleep(10.millis) >> ref.update(_ + batch.spans.size)
+  }
+
+  def refExporter(ref: Ref[IO, Chunk[CompletedSpan]]): SpanExporter[IO, Chunk] = new SpanExporter[IO, Chunk] {
+    def exportBatch(batch: Batch[Chunk]): IO[Unit] = ref.update(_ ++ batch.spans)
+  }
+
+  implicit def logger: Logger[IO] = Slf4jLogger.getLogger[IO]
+
+  val rps = 1000
+
+  it should "not block on complete" in forAll { (builder: CompletedSpan.Builder) =>
+    val test = for {
+      ref <- Ref.of[IO, Int](0)
+      random: Random[IO] <- Random.scalaUtilRandom[IO]
+      exporter = delayedExporter(ref)
+      res <- QueuedSpanCompleter[IO](
+        TraceProcess("completer-test"),
+        exporter,
+        CompleterConfig(bufferSize = 5, batchSize = 1)
+      ).use { completer =>
+        val randomRequestTime = for {
+          rand <- random.betweenDouble(0, 1)
+          _ <- IO.sleep((1.second * rand).asInstanceOf[FiniteDuration])
```

Comment:
Is this just jitter so we don't hammer it with everything at once?

Reply:
Yep, I believe so, I think @sgjbryan wanted to improve this a bit

Reply:
sounds good, I just removed it